chirpstack/src/maccommand/new_channel.rs

```diff
@@ -93,20 +93,17 @@ pub fn handle(
             return Err(anyhow!("Expected NewChannelReq"));
         };
 
-        if ans_pl.channel_freq_ok && ans_pl.dr_range_ok {
-            // Reset the error-counter.
-            ds.mac_command_error_count
-                .remove(&(lrwn::CID::NewChannelReq.to_u8() as u32));
-
-            ds.extra_uplink_channels.insert(
-                req_pl.ch_index as u32,
-                internal::DeviceSessionChannel {
-                    frequency: req_pl.freq,
-                    min_dr: req_pl.min_dr as u32,
-                    max_dr: req_pl.max_dr as u32,
-                },
-            );
+        // The device (positive/negative) acknowledged the channel. Mark it as configured.
+        ds.extra_uplink_channels.insert(
```

Comment:
This breaks in the case that we are updating a channel that has already been enabled. E.g.

* We want to update the frequency
* We want to update the data-rate range

By calling `.insert(...)` regardless of `_freq_ok=true`  and `dr_range_ok=true` we update the state of this channel in ChirpStack, and it will no longer reflect the state of the device because the device should keep its previous state:

![image](https://github.com/chirpstack/chirpstack/assets/165497/21def46d-5e73-4ff2-8730-ba0eff742375)

Reply:
I understand what you mean, but can we repurpose this mechanism to also indicate that the device (negatively) accepted the channel? I know that this is a weird way to think about it, but it tells Chirpstack that the channel was successfully conveyed to the device - although not accepted by the device (thus set to be disabled).

Otherwise, Chirpstack just keeps trying to set the channel over and over again (until the command fails), even if the device can never successfully acknowledge both the DR and channel frequency.